tests/cli/test_rasa_data.py

```diff
@@ -144,6 +152,415 @@ async def mock_from_importer(importer: TrainingDataImporter) -> Validator:
+            0.1,
+            1.0,
+            2,
+            0.5,
+        ),
+    ],
+)
+def test_rasa_data_augment_nlu_invalid_paths(
+    config_file: Text,
+    nlu_train_file: Text,
+    nlu_test_file: Text,
+    paraphrases_file: Text,
+    classification_report_file: Text,
+    min_paraphrase_sim_score: float,
+    max_paraphrase_sim_score: float,
+    augmentation_factor: float,
+    intent_proportion: float,
+):
+    tmp_path = rasa.utils.io.create_temporary_directory()
+    out_path = os.path.join(tmp_path, "augmentation_results")
+    os.makedirs(out_path)
+
+    data_root = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+
+    namespace = argparse.Namespace()
+    setattr(
+        namespace,
+        "nlu_training_data",
+        os.path.join(data_root, f"data/test_nlu_paraphrasing/{nlu_train_file}"),
+    )
+    setattr(
+        namespace,
+        "nlu_evaluation_data",
+        os.path.join(data_root, f"data/test_nlu_paraphrasing/{nlu_test_file}"),
+    )
+    setattr(
+        namespace,
+        "paraphrases",
+        os.path.join(data_root, f"data/test_nlu_paraphrasing/{paraphrases_file}"),
+    )
+    setattr(
+        namespace,
+        "nlu_classification_report",
+        os.path.join(
+            data_root, f"data/test_nlu_paraphrasing/{classification_report_file}"
+        ),
+    )
+    setattr(namespace, "out", out_path)
+    setattr(
+        namespace,
+        "config",
+        os.path.join(data_root, f"data/test_nlu_paraphrasing/{config_file}"),
+    )
+    setattr(namespace, "random_seed", 29306)
+    setattr(namespace, "intent_proportion", intent_proportion)
+    setattr(namespace, "augmentation_factor", augmentation_factor)
+    setattr(namespace, "min_paraphrase_sim_score", min_paraphrase_sim_score)
+    setattr(namespace, "max_paraphrase_sim_score", max_paraphrase_sim_score)
+
+    with pytest.raises(FileNotFoundException):
+        augment_nlu_data(namespace)
+
+
+@pytest.mark.parametrize(
+    "config_file, nlu_train_file, nlu_test_file, paraphrases_file, classification_report_file, "
+    "min_paraphrase_sim_score, max_paraphrase_sim_score, augmentation_factor, intent_proportion",
+    [
+        (
+            "config.yml",
+            "nlu_train.yml",
+            "nlu_test.yml",
+            "paraphrases.yml",
+            "nlu_classification_report_no_augmentation.json",
+            "0.1",
+            "1.0",
+            "2.0",
+            "0.5",
+        )
+    ],
+)
+def test_rasa_data_augment_nlu(
+    run: Callable[..., RunResult],
+    config_file: Text,
+    nlu_train_file: Text,
+    nlu_test_file: Text,
+    paraphrases_file: Text,
+    classification_report_file: Text,
+    min_paraphrase_sim_score: Text,
+    max_paraphrase_sim_score: Text,
+    augmentation_factor: Text,
+    intent_proportion: Text,
+):
+
+    tmp_path = rasa.utils.io.create_temporary_directory()
+    out_path = os.path.join(tmp_path, "augmentation_results")
+    os.makedirs(out_path)
```

Comment:
please use the `tmp_path` fixture. Ideally together with [pathlib](https://docs.python.org/3/library/pathlib.html) like our other tests.